### PR TITLE
DDCE-4956 - Fix WCAG Welsh Error Prefix + dep updates + code cleanup

### PR DESCRIPTION
**Pull request by lmugnai**
### Type of change
- [ ] Breaking change
- [ ] Non-breaking change
- [ ] Cosmetic change

### Description
Jira link: [Jira title](https://jira.tools.tax.service.gov.uk/browse/ARSSTB-151)

_Provide a short description of the change. For example: what parts of the code changed, is there any new behaviour...etc._

### Attachments or Screenshots
_Add any screenshots (if relevant) here_

### CODE REVIEW

File: app/viewmodels/ErrorMessageAwareness.scala [hunk 1]
```diff
@@ -25,5 +25,10 @@ trait ErrorMessageAwareness {
 
   def errorMessage(field: Field)(implicit messages: Messages): Option[ErrorMessage] =
     field.error
-      .map(err => ErrorMessage(content = Text(messages(err.message, err.args: _*))))
+      .map(
```
Comment (EdCope) [inline app/viewmodels/ErrorMessageAwareness.scala:28]: Fix working for most places.
Pages using component `YesNoRadios` still has a custom implementation. This prints Gwall/Error without the `:` like other pages.
https://github.com/hmrc/advance-valuation-rulings-frontend/blob/e6d1609fb03bfb921ae420398c5517e916c39f77/app/views/components/YesNoRadios.scala.html#L48-L60

**Review by EdCope — CHANGES_REQUESTED** [review]

File: app/controllers/AboutSimilarGoodsController.scala [hunk 2]
```diff
@@ -73,11 +74,12 @@ class AboutSimilarGoodsController @Inject() (
...
                 _              <- userAnswersService.set(updatedAnswers)
-              } yield saveDraft match {
-                case true  => Redirect(routes.DraftHasBeenSavedController.onPageLoad(draftId))
-                case false =>
+              } yield
+                if (saveDraft) {
+                  Redirect(routes.DraftHasBeenSavedController.onPageLoad(draftId))
+                } else {
                   Redirect(navigator.nextPage(AboutSimilarGoodsPage, mode, updatedAnswers))
-              }
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply app/controllers/AboutSimilarGoodsController.scala:80]: Yes, probably. But let's take a look at this on one of the other lesser priority tickets.

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
 resolvers += Resolver.url(
-  "HMRC-open-artefacts-ivy",
+  "HMRC-open-artefacts-ivy2",
   url("https://open.artefacts.tax.service.gov.uk/ivy2")
-)(Resolver.ivyStylePatterns)
+)(
+  Resolver.ivyStylePatterns
+)
 
 resolvers += "Typesafe Releases" at "https://repo.typesafe.com/typesafe/releases/"
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply project/plugins.sbt:12]: I'm can't see anywhere in conversations/confluence about removing (Resolver.ivyStylePatterns). I've removed the resolver from build.sbt though

File: run_all_tests.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+sbt clean scalafmtAll scalastyleAll compile coverage test coverageOff coverageReport dependencyUpdates
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply run_all_tests.sh:2]: Done

File: project/plugins.sbt (hunk 3 above)
[the remark this answers is not in the document]
Comment (EdCope) [reply project/plugins.sbt:12]: Sorry was referring to the typesafe resolver

**Review by EdCope — APPROVED** [review]

**Pull request merged**